### PR TITLE
 Replace some uses of deprecated os (mostly os::args) functions

### PR DESCRIPTION
**Pull request by nagisa**
This commit mostly replaces some of the uses of os::args with env::args.

This, for obvious reasons is based on top of #22400. Do not r+ before that lands.

### CODE REVIEW

File: src/test/bench/rt-messaging-ping-pong.rs [hunk 1]
```diff
@@ -63,17 +63,15 @@ fn ping_pong_bench(n: uint, m: uint) {
...
     let n = if args.len() == 3 {
-        args[1].parse::<uint>().unwrap()
+        args.nth(1).unwrap().parse::<usize>().unwrap()
     } else {
         10000
     };
 
     let m = if args.len() == 3 {
-        args[2].parse::<uint>().unwrap()
+        args.next().unwrap().parse::<usize>().unwrap()
```
Comment (nagisa) [inline src/test/bench/rt-messaging-ping-pong.rs]: This one is not as obvious as it was before, which possibly makes refactoring this file a little bit harder if we ever introduce breaking changes to `env::args` in the future.

Not sure how important it is, though.

**Comment by nikomatsakis** [comment]
Looks good to me!

**Comment by Manishearth** [comment]
I'm going ahead and rolling this one up, ignoring the other PR, since in the rollup they'll all be together anyway. Let me know if I shouldn't.

**Comment by nagisa** [comment]
OK, I closed the other one.

**Comment by Manishearth** [comment]
Thanks. I'll be sure to let you know if the rollup fails and it needs reopening :)

**Comment by nagisa** [comment]
https://github.com/rust-lang/rust/pull/22475 didn’t auto-close this somewhy…

**Comment by Manishearth** [comment]
Probably  because you pushed more changes here (which Alex may have included in a different form)

**Pull request closed without merging**